File: lib/fixtures/src/main/java/bankaccount/query/CurrentBalanceQueries.java
package bankaccount.query;

import org.axonframework.messaging.responsetypes.ResponseTypes;
import org.axonframework.queryhandling.QueryGateway;

import java.util.concurrent.CompletableFuture;

public class CurrentBalanceQueries {
  private final QueryGateway queryGateway;

  public CurrentBalanceQueries(QueryGateway queryGateway) {
    this.queryGateway = queryGateway;
  }

  public CompletableFuture<CurrentBalanceResult> findByAccountId(String accountId) {
    return queryGateway.query(new CurrentBalanceQuery(accountId), ResponseTypes.instanceOf(CurrentBalanceResult.class));
  }

  public CompletableFuture<CurrentBalanceResultList> findAll() {
    return queryGateway.query(FindAllQuery.getINSTANCE(), ResponseTypes.instanceOf(CurrentBalanceResultList.class));
  }

}

File: _bak/lib/fixture-application/src/main/kotlin/bankaccount/query/CurrentBalanceQueries.kt
package bankaccount.query

import org.axonframework.messaging.responsetypes.ResponseTypes
import org.axonframework.queryhandling.QueryGateway
import java.util.concurrent.CompletableFuture

class CurrentBalanceQueries(queryGateway: QueryGateway) {
  private val queryGateway: QueryGateway = queryGateway

  fun findByAccountId(accountId: String): CompletableFuture<CurrentBalanceResult> {
    return queryGateway.query(
      CurrentBalanceQuery(accountId),
      ResponseTypes.instanceOf(CurrentBalanceResult::class.java)
    )
  }

  fun findAll(): CompletableFuture<CurrentBalanceResultList> {
    return queryGateway.query(
      FindAllQuery.INSTANCE,
      ResponseTypes.instanceOf(CurrentBalanceResultList::class.java)
    )
  }
}
